### DISCUSSION TITLE
Using Frigate with Traefik and Home Assistant (docker)

### DISCUSSION THREAD

**Discussion by shred86** [blakeblackshear/frigate#6337] [Ask A Question] [answered]
My setup is a mini PC (Debian OS) running Frigate, Home Assistant (not Hass OS) and several other services in docker containers. I also run the Frigate HA integration which connects via <local_ip>:5000. I recently setup Traefik and would like to use it to access Frigate so I can put some basic authentication in front of it, meaning I would disable direct port access to <local_ip>:5000. I'm assuming this would break the Frigate HA integration, so I'm not quite sure how to go about this (or if it's even possible). Any tips on how to achieve this? Thanks.

**Comment by blakeblackshear** [comment, marked as answer]
See here: https://github.com/blakeblackshear/frigate/discussions/6250#discussioncomment-5726251

**Reply by shred86** [reply]
Ah perfect, thank you!

For anyone else that might be looking into this, you can map your Frigate docker container to the gateway of the docker network, which I didn't even realize was possible. From there, just use the IP address of the docker gateway with port 5000 to connect to Frigate from the Home Assistant Frigate integration. Basically Home Assistant will connect to Frigate via the internal Docker network.

````yaml
  frigate:
    ...
    ports:
      - "172.19.0.1:5000:5000"
````

**Reply by hagak** [reply]
Another option is if you have both containers in the same bridge network and you give your containers a name, then you can reference them via container name.  IE give frigate a container_name: frigate, then you can access frigate via frigate:5000. This way if your docker networks change for some reason this config would not need to change.

**Reply by blakeblackshear** [reply]
> Another option is if you have both containers in the same bridge network and you give your containers a name, then you can reference them via container name.

This doesn't work when you have some containers like home assistant and esphome running in host network mode. My example compose shows how to have a reverse proxy in bridge mode talk to a container in host mode and a container in host mode talk to a container in bridge mode.

**Reply by shred86** [reply]
Not sure what happened but I can't seem to stream my cameras anymore through Home Assistant. The Frigate integration loads fine and all devices/entities show up correctly to include the camera snapshots. However, when I attempt to view a live feed, I just get a blank screen. In the HA logs, I see multiple of the following error for all my cameras:

````Error from stream worker: Error opening stream (HTTP_NOT_FOUND, Server returned 404 Not Found) rtsp://172.19.0.1:8554/back````

Nothing has changed with my docker-compose.yaml file, and I've tried restarting everything but still no luck. It's strange because the HA Frigate integration is clearly able to communicate with Frigate, it's just the live streams that are not working.

**Reply by shred86** [reply]
For anyone that comes across this, not sure why this didn't work for me the first time but after some [discussion on reddit](https://www.reddit.com/r/frigate_nvr/comments/1729p93/frigate_docker_container_behind_traefik_basicauth/), it appears I'm able to just use the Frigate container IP address on the docker network for the Home Assistant Frigate integration. I thought I tried this before and it didn't work, which I thought was contributed to using host networking for my Home Assistant container (same host as Frigate), but it definitely works.

So as a recap, I'm running Home Assistant and Frigate on the same docker host. Home Assistant is running in network host mode, and Frigate is behind Traefik for basicauth. I connect the Home Assistant Frigate integration to the Frigate container using the Frigate container IP address on the docker network. I don't have any ports exposed in my docker-compose.yml for Frigate.

You will also probably want to set a static IP address for your docker container or else it could change when you restart Docker, thus breaking the connection between Home Assistant and Frigate. You can do that by adding this to your Frigate container service in the docker-compose.yml:

````yaml

    networks:
      proxy:
        ipv4_address: 172.19.0.102

````

proxy is the name of my docker network for Traefik. I just choose a high number IP address since I'll never have more than a dozen containers, there shouldn't be a conflict.

**Comment by shred86** [comment]
One issue I'm noticing is with using Traefik's basicAuth, when I click different pages within the Frigate UI, it often asks me to reauthenticate (not always, but usually). I don't get this issue with using basicAuth on my Traefik dashboard, so I'm not sure what's going on. I'm simply running a label on my Frigate container `"traefik.http.routers.frigate.middlewares=traefik-auth"`, with the middleware being created on my Traefik container label, `"traefik.http.middlewares.traefik-auth.basicauth.users=user:pass"`.

**Comment by ptruman** [comment]
The issue will be likely that you will need to setup traefik entrypoints for ALL the ports frigate uses, because when HASS (or other items) use the name (URL) provided, traefik will have only opened that one port going to 5000 - not all the others.

I've put frigate on a docker macvlan IP so it's dedicated, and am about to move it behind traefik

**Reply by shred86** [reply]
I'm assuming you're running the network in host mode? I'm curious if you still run into a similar issue with that setup. I'm not sure which other ports to setup with Frigate. The only other port I see mentioned in the docs is 8555 for WebRTC.

**Reply by NickM-27** [reply]
The installation docs show all the ports that are used (externally) and what they're for.

**Comment by PrplHaz4** [comment]
> Not sure what happened but I can't seem to stream my cameras anymore through Home Assistant. The Frigate integration loads fine and all devices/entities show up correctly to include the camera snapshots. However, when I attempt to view a live feed, I just get a blank screen. In the HA logs, I see multiple of the following error for all my cameras:
> 
> `Error from stream worker: Error opening stream (HTTP_NOT_FOUND, Server returned 404 Not Found) rtsp://172.19.0.1:8554/back`
> 
> Nothing has changed with my docker-compose.yaml file, and I've tried restarting everything but still no luck. It's strange because the HA Frigate integration is clearly able to communicate with Frigate, it's just the live streams that are not working.

Did you check to make sure your docker gateway is still on that IP?  Changes to docker networks can result in the gateway ip shifting around, so it may not be at `172.19.0.1` anymore. You might be able to use `host.docker.internal` or `gateway.docker.internal` to reach services on the host more reliably.
https://docs.docker.com/desktop/networking/#i-want-to-connect-from-a-container-to-a-service-on-the-host
